proxy/request_handler.go

```diff
@@ -334,6 +336,15 @@ func (d *RequestHandler) InitializeAuthnSession(r *http.Request, rl *rule.Rule)
 
 	session := &authn.AuthenticationSession{
 		Subject: "",
+		Request: *r.Clone(r.Context()),
+	}
+
+	if r.Body != nil {
+		bodyBytes, _ := ioutil.ReadAll(r.Body)
```

Comment:
I don't quite remember but I think we're doing this already somewhere? While this is functionally doing the same here's a version that uses io.TreeReader to do that:

```
func pipeRequestBody(r *http.Request, w io.Writer) error {
	var body bytes.Buffer
	_, err := io.Copy(w, io.TeeReader(r.Body, &body))
	r.Body = ioutil.NopCloser(&body)
	return err
}

```

Reply:
Could you take a look and if you like the current implementation?